hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/TypeMatcher.kt

```diff
@@ -0,0 +1,15 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch.api
+
+import kotlin.reflect.KClass
+
+sealed interface TypeMatcher<S> {
```

Comment:
The TypeMatcher is now generic and we use the supported type when validating the matcher supports the requested attribute (e.g. if attribute "firstName" is used in anything other than a `TypeMatcher<String>` then validation will fail.